packages/ui/src/memberships/components/MemberListFilters/MemberListFilters.tsx

```diff
@@ -0,0 +1,155 @@
+import React, { ChangeEvent, useReducer } from 'react'
+import styled from 'styled-components'
+
+import { ButtonPrimary } from '../../../common/components/buttons/Buttons'
+import { InputComponent, InputText, TogglableIcon } from '../../../common/components/forms'
+import { FounderMemberIcon, VerifiedMemberIcon } from '../../../common/components/icons'
+import { SimpleSelect } from '../../../common/components/selects'
+import { Colors } from '../../../common/constants'
+
+export interface MemberListFilter {
+  search: string
+  role: string | null
+  concil: boolean | null
+  onlyVerified: boolean
+  onlyFounder: boolean
+}
+type FilterKey = keyof MemberListFilter
+
+type Clear = { type: 'clear' }
+type Change<K extends FilterKey = FilterKey> = K extends FilterKey // Use a conditional type in order to distribue the union type
+  ? { type: 'change'; field: K; value: MemberListFilter[K] }
+  : never
+type Action = Clear | Change
+
+const filterReducer = (filters: MemberListFilter, action: Action): MemberListFilter => {
+  switch (action.type) {
+    case 'clear':
+      return MemberListEmptyFilter
+
+    case 'change':
+      return { ...filters, [action.field]: action.value }
+  }
+  return filters
+}
+
+const CouncilOpts = {
+  All: null,
+  Yes: true,
+  no: false,
+}
+
+export const MemberListEmptyFilter: MemberListFilter = {
+  search: '',
+  role: null,
+  concil: CouncilOpts.All,
+  onlyVerified: false,
+  onlyFounder: false,
+}
+
+interface MemberListFiltersProps {
+  roles: { [k: string]: string }
+  onApply?: (value: MemberListFilter) => void
+}
+export const MemberListFilters = ({ roles, onApply }: MemberListFiltersProps) => {
+  const [filters, dispatch] = useReducer(filterReducer, MemberListEmptyFilter)
+  const { search, role, concil, onlyVerified, onlyFounder } = filters
+
+  return (
+    <MemberListFiltersContainer>
+      <ClearBtn
+        onClick={() => {
+          dispatch({ type: 'clear' })
+          onApply?.(MemberListEmptyFilter)
+        }}
+      >
+        Clear all Filters
+      </ClearBtn>
+
+      <Fields>
+        <InputComponent icon={<>🔍</>} /* TODO replace by the actual icon */>
```

Comment:
Hello @jodator thanks for the review :). I fix some of the CSS but there is still some subtle movement when interacting with the select (sometime), but then I realized it was caused by fonts being reloaded in Storybook after some DOM updates. I don’t think that’s what you experienced since there was a real positioning problem with the previous code.